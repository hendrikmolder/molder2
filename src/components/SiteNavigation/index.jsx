import React from "react"
import PropTypes from 'prop-types'
import { Link } from 'gatsby'
import styled from 'styled-components'

const NavigationContainer = styled.div`
    position: absolute;
    top: 0;
    width: 100%;
    padding: 20px 30px 10px;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
`

const StyledLink = styled(Link)`
    font-size: 14px;
    font-weight: 500;
    margin: 0 10px 0;
    color: rgb(132, 132, 132);

    :hover {
        text-decoration: underline;
        color: rgb(132, 132, 132);
    }

`

const SiteNavigation = (props) => {
    const { items } = props

    return (
        <NavigationContainer>
            <StyledLink to="/" activeStyle={{ color: 'black' }}>home</StyledLink>
            { items.map((item, key) => {
                const { name, href } = item

                return (
                    <StyledLink
                        to={href}
                        key={key}
                        activeStyle={{ color: 'black' }}
                        partiallyActive={true}
                    >
                        {name}
                    </StyledLink>
                )
            })}
        </NavigationContainer>
    )
}

SiteNavigation.propTypes = {
    items: PropTypes.array.isRequired,
}

export default SiteNavigation