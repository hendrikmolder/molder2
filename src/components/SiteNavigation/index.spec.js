import React from 'react'
import renderer from 'react-test-renderer'

import SiteNavigation from './index'

const TEST_MENU_ITEMS = [
    { name: "bio", href: "/" },
    { name: "blog", href: "/blog" },
]

describe('SiteNavigation', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<SiteNavigation items={TEST_MENU_ITEMS} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})