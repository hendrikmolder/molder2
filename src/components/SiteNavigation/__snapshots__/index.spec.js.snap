// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteNavigation renders correctly 1`] = `
<div
  className="sc-bdVaJa ekuuOW"
>
  <a
    href="/"
  >
    home
  </a>
  <a
    href="/"
    partiallyActive={true}
  >
    bio
  </a>
  <a
    href="/blog"
    partiallyActive={true}
  >
    blog
  </a>
</div>
`;
